In [1]:
import pandas as pd, numpy as np
import yaml
import warnings
warnings.filterwarnings('ignore')

!date
!whoami

Thu Apr 16 15:12:31 PDT 2020
yaqiw


## This notebook is to compare yaml random_seeds to output.hdf random_seeds

In [2]:
def checkRandomSeed(filePath, country):
    #load output data
    output = pd.read_hdf(filePath.get(country) + '/output.hdf')
    outputSeedSize = output.groupby('input_draw').random_seed.count()
    #load yaml data
    with open(filePath.get(country) + '/keyspace.yaml', 'r') as f:
        yamlData = yaml.load(f.read())
    yamlSeedSize = len(yamlData['random_seed'])
    #compare random seeds
    if len(outputSeedSize[outputSeedSize != yamlSeedSize]) != 0:
        print(f'Attention: {country} has unmatched random_seed counts!')
        print(f'Following are details, random_seed counts from yaml are {yamlSeedSize}')
        print(outputSeedSize[outputSeedSize != yamlSeedSize])
        print('\n')
    else: 
        print(f'{country} has matched random_seed counts')

In [3]:
countries = ['Ethiopia', 'India', 'Nigeria']

### 1. check random_seed counts for model 3

In [4]:
result_dir = '/ihme/costeffectiveness/results/vivarium_conic_lsff/v3.0_paf_of_one/'
filePath = {
    'Ethiopia': result_dir + 'ethiopia/2020_03_17_14_59_54',
    'India': result_dir + 'india/2020_03_17_15_00_02',
    'Nigeria': result_dir + 'nigeria/2020_03_17_15_00_10'
}

In [5]:
for country in countries:
    checkRandomSeed(filePath, country)

Ethiopia has matched random_seed counts
India has matched random_seed counts
Nigeria has matched random_seed counts


### 2. check random_seed counts for model 4

In [6]:
result_dir = '/ihme/costeffectiveness/results/vivarium_conic_lsff/v4.0_folic_acid_fortification/'
filePath = {
    'Ethiopia': result_dir + 'ethiopia/2020_03_18_23_04_26',
    'India': result_dir + 'india/2020_03_18_23_04_36',
    'Nigeria': result_dir + 'nigeria/2020_03_18_23_04_42'
}

In [7]:
for country in countries:
    checkRandomSeed(filePath, country)

Attention: Ethiopia has unmatched random_seed counts!
Following are details, random_seed counts from yaml are 40
input_draw
21.0     80
29.0     80
55.0     80
78.0     80
155.0    80
223.0    80
232.0    80
254.0    80
307.0    80
357.0    80
394.0    80
417.0    80
482.0    80
514.0    80
524.0    80
533.0    80
602.0    80
620.0    80
629.0    80
650.0    80
674.0    80
680.0    80
733.0    80
829.0    80
946.0    80
Name: random_seed, dtype: int64


Attention: India has unmatched random_seed counts!
Following are details, random_seed counts from yaml are 40
input_draw
21.0     80
29.0     80
55.0     80
78.0     80
155.0    80
223.0    80
232.0    80
254.0    80
307.0    80
357.0    80
394.0    80
417.0    80
482.0    80
514.0    80
524.0    80
533.0    80
602.0    80
620.0    80
629.0    80
650.0    80
674.0    80
680.0    80
733.0    80
829.0    80
946.0    80
Name: random_seed, dtype: int64


Attention: Nigeria has unmatched random_seed counts!
Following are details, random_seed

In [8]:
def checkRandomSeed(filePath, country):
    #load output data
    output = pd.read_hdf(filePath.get(country) + '/output.hdf')
    outputSeedSize = output.groupby('input_draw').random_seed.count()
    #load yaml data
    with open(filePath.get(country) + '/keyspace.yaml', 'r') as f:
        yamlData = yaml.load(f.read())
    yamlSeedSize = len(yamlData['random_seed'])
    #compare random seeds
    if len(outputSeedSize[outputSeedSize != yamlSeedSize]) != 0:
        print(f'Attention: {country} has unmatched random_seed counts!')
        print(f'Following are details, random_seed counts from yaml are {yamlSeedSize}')
        print(outputSeedSize[outputSeedSize != yamlSeedSize])
        print('\n')
    else: 
        print(f'{country} has matched random_seed counts')

### 3. check random_seed counts for model 5

In [9]:
result_dir = '/ihme/costeffectiveness/results/vivarium_conic_lsff/v5.0_vitamin_a_fortification/'
filePath = {
    'Ethiopia': result_dir + 'ethiopia/2020_03_26_08_33_10',
    'India': result_dir + 'india/2020_03_26_21_28_29',
    'Nigeria': result_dir + 'nigeria/2020_03_24_23_27_55'
}

In [10]:
for country in countries:
    checkRandomSeed(filePath, country)

Attention: Ethiopia has unmatched random_seed counts!
Following are details, random_seed counts from yaml are 40
input_draw
21.0     120
29.0     120
55.0     120
78.0     120
155.0    120
223.0    120
232.0    120
254.0    120
307.0    120
357.0    120
394.0    120
417.0    120
482.0    120
514.0    120
524.0    120
533.0    120
602.0    120
620.0    120
629.0    120
650.0    120
674.0    120
680.0    120
733.0    120
829.0    120
946.0    120
Name: random_seed, dtype: int64


Attention: India has unmatched random_seed counts!
Following are details, random_seed counts from yaml are 40
input_draw
21.0     120
29.0     120
55.0     120
78.0     120
155.0    120
223.0    120
232.0    120
254.0    120
307.0    120
357.0    120
394.0    120
417.0    120
482.0    120
514.0    120
524.0    120
533.0    120
602.0    120
620.0    120
629.0    120
650.0    120
674.0    120
680.0    120
733.0    120
829.0    120
946.0    120
Name: random_seed, dtype: int64


Attention: Nigeria has unmatched rand

### check why random_seed counts number is 120, take Ethiopia from model 5 for example

In [11]:
df = pd.read_hdf(filePath.get('Ethiopia') + '/output.hdf')

In [12]:
df.input_draw.unique()

array([829., 155., 394., 533.,  55., 232., 524., 602.,  78., 514., 254.,
       307., 417., 733., 680., 223., 482.,  29., 650., 629., 357., 946.,
        21., 620., 674.])

In [13]:
df.loc[df['input_draw'] == 21].random_seed.value_counts()

205.0    3
346.0    3
199.0    3
228.0    3
215.0    3
6.0      3
145.0    3
280.0    3
45.0     3
337.0    3
139.0    3
306.0    3
38.0     3
258.0    3
114.0    3
43.0     3
92.0     3
389.0    3
76.0     3
377.0    3
388.0    3
203.0    3
243.0    3
16.0     3
88.0     3
245.0    3
66.0     3
395.0    3
169.0    3
69.0     3
87.0     3
235.0    3
134.0    3
230.0    3
55.0     3
207.0    3
35.0     3
133.0    3
185.0    3
129.0    3
Name: random_seed, dtype: int64